src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py

```diff
@@ -409,3 +409,92 @@ def similarity_graph(self, k=5, radius=None, include_self_edges=False,
         +----------+----------+----------------+------+
         """
         return self.similarity_model.similarity_graph(k, radius, include_self_edges, output_type, verbose)
+
+    def export_coreml(self, filename, reference_data):
+        """
+        Save the model in Core ML format.
+
+        See Also
+        --------
+        save
+
+        Examples
+        --------
+        >>> model.export_coreml('myModel.mlmodel')
+        """
+        import numpy as _np
+        import coremltools as _cmt
+        from coremltools.models import datatypes as _datatypes, neural_network as _neural_network
+        from .._mxnet_to_coreml import _mxnet_converter
+        from turicreate.toolkits import _coreml_utils
+
+        input_name = self.feature_extractor.data_layer
+        output_name = 'distance'
+
+        output_size = reference_data.num_rows()
+
+        input_features = [(input_name, _datatypes.Array(*(self.input_image_shape)))]
+        output_features = [(output_name, _datatypes.Array(*(output_size, )))]
+
+        builder = _neural_network.NeuralNetworkBuilder(
+            input_features, output_features, mode='classifier')
+
+        # Convert the feature exctraction network
+        mx_feature_extractor = self.feature_extractor._get_mx_module(
+            self.feature_extractor.ptModel.mxmodel,
+            self.feature_extractor.data_layer,
+            self.feature_extractor.feature_layer,
+            self.feature_extractor.context,
+            self.input_image_shape
+        )
+
+        batch_input_shape = (1, ) + self.input_image_shape
+        _mxnet_converter.convert(mx_feature_extractor, mode=None,
+                                 input_shape={input_name: batch_input_shape},
+                                 builder=builder, verbose=False)
+
+        # Calculate the euclidean distance between the newly extracted query features
```

Comment:
I'm having a really difficult time understanding the code that ensues. Any additional comments you can leave here would be very helpful.